### PR TITLE
Issue #7848: SuppressWarningsHolder should not report violations

### PR DESCRIPTION
**Pull request by AmrDeveloper**
Issue #7848 

Diff Report: https://amrdeveloper.github.io/checkstyle-reports/7848/

### CODE REVIEW

**Comment by AmrDeveloper** [comment]
@rnveach we need UT to cover default case in getAcceptableParent https://github.com/checkstyle/checkstyle/issues/7848#issuecomment-602039350 to pass the CI

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/SuppressWarningsHolder.java [hunk 1]
```diff
@@ -441,6 +441,7 @@ private static DetailAST getAcceptableParent(DetailAST child) {
             case TokenTypes.DOT:
             case TokenTypes.ANNOTATION:
             case TokenTypes.MODIFIERS:
+            case TokenTypes.RESOURCE:
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/SuppressWarningsHolder.java]: We need to remove all functionality that is logging of violations. 
Please remove line with logging of violation, remove message, all locations of annotations is valid

Comment (AmrDeveloper) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/SuppressWarningsHolder.java]: you mean delete getAcceptableParent method and accept all parent type and delete log method right ?

Comment (AmrDeveloper) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/SuppressWarningsHolder.java]: @romani Done

File: src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressionXpathFilter.java [hunk 2]
```diff
@@ -116,6 +116,9 @@
  * RegexpSinglelineJava
  * </li>
  * <li>
+ * SuppressWarningsHolder
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressionXpathFilter.java:119]: please remove it from here.
this module do not report any violations now, so it should not be there.
Module will never report violations, Xpath will never be able to suppress violations from it. 

UT might be complaining on absent module, so we need to update UT to skip skip module from consideration as it has no violations, so no need to have support of Xpath for it.

Comment (AmrDeveloper) [reply src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressionXpathFilter.java]: @romani Done

File: src/test/java/com/puppycrawl/tools/checkstyle/internal/XpathRegressionTest.java [hunk 3]
```diff
@@ -155,6 +155,7 @@
             "SuperClone",
             "SuperFinalize",
             "SuppressWarnings",
+            "SuppressWarningsHolder",
```
Comment (rnveach) [inline src/test/java/com/puppycrawl/tools/checkstyle/internal/XpathRegressionTest.java:158]: This should be moved to something else as the check will never have xpath support. It doesn't create violations.

I would create a new field for it that specifically says they will not have xpath support ever, and give a reason why for this check.

Comment (AmrDeveloper) [reply src/test/java/com/puppycrawl/tools/checkstyle/internal/XpathRegressionTest.java:158]: @rnveach Thanks, You are right
Once you create the new field I will rebase and move it to the new field

Comment (rnveach) [reply src/test/java/com/puppycrawl/tools/checkstyle/internal/XpathRegressionTest.java:158]: @AmrDeveloper I was referring to as if I was working the PR myself. You should make the changes as part of your PR.

Comment (romani) [reply src/test/java/com/puppycrawl/tools/checkstyle/internal/XpathRegressionTest.java:158]: As part of this PR, you need to update UT to skip such module at all.

File: src/test/java/com/puppycrawl/tools/checkstyle/internal/XpathRegressionTest.java [hunk 4]
```diff
@@ -160,6 +160,12 @@
             "VisibilityModifier"
     ));
 
+    // Checks that will never have xpath support ever because they not report violations
+    // till https://github.com/checkstyle/checkstyle/issues/7848
+    private static final Set<String> NO_VIOLATION_CHECKS = new HashSet<>(Collections.singletonList(
```
Comment (romani) [inline src/test/java/com/puppycrawl/tools/checkstyle/internal/XpathRegressionTest.java:165]: NO_VIOLATION_MODULES

All Checks must have violations.

Comment (AmrDeveloper) [reply src/test/java/com/puppycrawl/tools/checkstyle/internal/XpathRegressionTest.java]: Done

File: src/test/java/com/puppycrawl/tools/checkstyle/internal/XpathRegressionTest.java [hunk 5]
```diff
@@ -160,6 +160,12 @@
             "VisibilityModifier"
     ));
 
+    // Modules that will never have xpath support ever because they not report violations
+    // till https://github.com/checkstyle/checkstyle/issues/7848
```
Comment (romani) [inline src/test/java/com/puppycrawl/tools/checkstyle/internal/XpathRegressionTest.java:164]: please remove comment with "till".
"till" means .... when such issue become closed, we can remove code below.
But it is the same issue as you fix. Such block of code will exists till we remove SuppressWarningsHolder from repo, we do not have issue on this for now, it is far future.

File: src/test/java/com/puppycrawl/tools/checkstyle/checks/SuppressWarningsHolderTest.java [hunk 6]
```diff
@@ -259,9 +259,7 @@ public void testIsSuppressedWithAllArgument() throws Exception {
     public void testAnnotationInTry() throws Exception {
         final Configuration checkConfig = createModuleConfig(SuppressWarningsHolder.class);
 
-        final String[] expected = {
-            "11:14: " + getCheckMessage(SuppressWarningsHolder.MSG_KEY),
```
Comment (romani) [inline src/test/java/com/puppycrawl/tools/checkstyle/checks/SuppressWarningsHolderTest.java:263]: please remove MSG_KEY from module, it should not be used any more.

File: src/test/java/com/puppycrawl/tools/checkstyle/internal/AllChecksTest.java [hunk 7]
```diff
@@ -488,6 +488,11 @@ public void testAllCheckstyleChecksHaveMessage() throws Exception {
         for (Class<?> module : CheckUtil.getCheckstyleChecks()) {
             final String name = module.getSimpleName();
 
+            // hidden exception messages
+            if ("SuppressWarningsHolder".equals(name)) {
```
Comment (romani) [inline src/test/java/com/puppycrawl/tools/checkstyle/internal/AllChecksTest.java:492]: please remove comment, I do not understand it. Exception will fail whole Checkstyle execution, it is not validation, any Check can throw exception. Or make comment `// No messages in just module`

Please try to remove `SuppressWarningsHolder` from `getCheckstyleChecks` return, if there a lot of other changes(please share them here) are required then lets keep it and we will create new issue on this and address separately.

Comment (AmrDeveloper) [reply src/test/java/com/puppycrawl/tools/checkstyle/internal/AllChecksTest.java:492]: @romani  I updated the comment to // No messages in just module

and I am waiting for Travis to rebase if it's failed

in `getCheckstyleChecks` we can add one more filter to remove `SuppressWarningsHolder` from the result 

    public static Set<Class<?>> getCheckstyleChecks() throws IOException {
            final ClassLoader loader = Thread.currentThread().getContextClassLoader();
            final String packageName = "com.puppycrawl.tools.checkstyle";
            return getCheckstyleModulesRecursive(packageName, loader).stream()
                    .filter(ModuleReflectionUtil::isCheckstyleTreeWalkerCheck)
                    .filter(aClass -> "SuppressWarningsHolder".equals(aClass.getName()))
                    .collect(Collectors.toSet());
    }

**Comment by romani** [comment]
travis is failed during site generation during connection to w3c web site. restarted.

File: src/test/java/com/puppycrawl/tools/checkstyle/internal/AllChecksTest.java (hunk 7 above)
Comment (romani) [reply src/test/java/com/puppycrawl/tools/checkstyle/internal/AllChecksTest.java:492]: Ok, lets do this in separate PR , new issue https://github.com/checkstyle/checkstyle/issues/8121 .
You are welcome to fix it as you know all details.

**Pull request merged**